src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,103 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private final float loadFactor;
+    private int size;
+    private int capacity;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY, DEFAULT_LOAD_FACTOR);
+    }
+
+    public MyHashMap(int capacity, float loadFactor) {
+        this.capacity = capacity;
+        this.loadFactor = loadFactor;
+        this.table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        resize();
+        int index = getIndex(key);
+        Node node = table[index];
+        while (node != null) {
+            if ((node.key != null && node.key.equals(key))
+                    || node.key == null && key == null) {
+                node.value = value;
+                return;
+            }
+            node = node.next;
+        }
+        Node<K,V> newNode = new Node<>(key,value);
+        newNode.next = table[index];
+        table[index] = newNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int index = getIndex(key);
+        Node<K,V> node = table[index];
+        while (node != null) {
+            if ((node.key != null && node.key.equals(key))
+                    || node.key == null && key == null) {
+                return node.value;
+            }
+            node = node.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void resize() {
+        if (size > getThreshold()) {
+            capacity = capacity * CAPACITY_MULTIPLIER;
+            Node[] newTable = new Node[capacity];
```

Comment:
It's better not to use Node in raw type
 ```suggestion
            Node<K, V>[] newTable = new Node[capacity];
```